Neos.Flow/Classes/ObjectManagement/ObjectManager.php

```diff
@@ -403,24 +405,37 @@ public function getLazyDependencyByHash($hash, &$propertyReferenceVariable)
      * Internally used by the injectProperties method of generated proxy classes.
      *
      * @param string $hash An md5 hash over the code needed to actually build the dependency instance
-     * @param string &$propertyReferenceVariable A first variable where the dependency needs to be injected into
+     * @param mixed &$propertyReferenceVariable A first variable where the dependency needs to be injected into
      * @param string $className Name of the class of the dependency which eventually will be instantiated
-     * @param \Closure $builder An anonymous function which creates the instance to be injected
-     * @return DependencyProxy
+     * @param Closure $builder An anonymous function which creates the instance to be injected
+     * @return DependencyProxy | null
      */
-    public function createLazyDependency($hash, &$propertyReferenceVariable, $className, \Closure $builder): DependencyProxy
+    public function createLazyDependency(string $hash, &$propertyReferenceVariable, string $className, Closure $builder): ?DependencyProxy
     {
-        $this->dependencyProxies[$hash] = new DependencyProxy($className, $builder);
+        // Trigger auto-loading of the original class, because any generated lazy proxy will be
+        // contained in the same file and the auto-loader does not know how to find a lazy proxy:
+        class_exists($className) || interface_exists($className);
+        $lazyProxyClassName = $className . '_LazyProxy';
+
+        // If proxy classes were partially flushed in Development context, it might happen that
+        // the original class $className was already loaded by the ConfigurationBuilder. In that
+        // case, the proxy class for $className cannot be loaded any more by the ProxyClassLoader,
+        // so we need to fall back to eager dependency injection.
+        if (!class_exists($lazyProxyClassName)) {
+            return null;
+        }
+
+        $this->dependencyProxies[$hash] = call_user_func_array([$lazyProxyClassName, '_createDependencyProxy'], [$className, $builder]);
```

Comment:
```suggestion
        $this->dependencyProxies[$hash] = $lazyProxyClassName::_createDependencyProxy($className, $builder);
```

Reply:
Makes sense – did you test this?

Reply:
will ^^
but made something like this yesterday syntax wise and it worked for a static method on a classname variable

Reply:
At least the demo site works fine with this, and the tests pass, too.